content/master/operations/operation.md

```diff
@@ -0,0 +1,580 @@
+    requirements:
+      requiredResources:
+      - requirementName: app-deployment
+        apiVersion: apps/v1
+        kind: Deployment
+        name: my-app
+        namespace: production
+      - requirementName: app-service
+        apiVersion: v1
+        kind: Service
+        name: my-app-service
+        namespace: production
+    input:
+      apiVersion: fn.crossplane.io/v1beta1
+      kind: ProcessorInput
+      action: upgrade
+```
+
+Functions access these resources through the standard request structure:
+
+```python
+from crossplane.function import request, response
+
+def operate(req, rsp):
+    # Access required resources
+    deployment = request.get_required_resource(req, "app-deployment")
+    service = request.get_required_resource(req, "app-service")
+    
+    if not deployment or not service:
+        response.set_output(rsp, {"error": "Required resources not found"})
+        return
+    
+    # Process the resources
+    new_replicas = deployment["spec"]["replicas"] * 2
+    
+    # Return updated resources with full GVK and metadata for server-side apply
+    rsp.desired.resources["app-deployment"].resource.update({
+        "apiVersion": "apps/v1",
+        "kind": "Deployment",
+        "metadata": {
+            "name": deployment["metadata"]["name"],
+            "namespace": deployment["metadata"]["namespace"]
+        },
+        "spec": {"replicas": new_replicas}
+    })
+```
+
+## Status and monitoring
+
+Operations provide rich status information:
+
+```yaml
+status:
+  conditions:
+  - type: Synced
+    status: "True"
+    reason: ReconcileSuccess
+  - type: Succeeded
+    status: "True"
+    reason: PipelineSuccess
+  - type: ValidPipeline
+    status: "True"
+    reason: ValidPipeline
+  failures: 1  # Number of retry attempts
+  pipeline:
+  - step: create-backup
+    output:
+      backupId: "backup-20240115-103000"
+      size: "2.3GB"
+  appliedResourceRefs:
+  - apiVersion: "v1"
+    kind: "Secret"
+    namespace: "production"
+    name: "backup-secret"
+  - apiVersion: "apps/v1" 
+    kind: "Deployment"
+    name: "updated-deployment"
+```
+
+**Key status fields:**
+- **Conditions**: Standard Crossplane conditions (Synced) and Operation-specific conditions:
+  - **Succeeded**: `True` when the operation completed successfully, `False` when it failed
+  - **ValidPipeline**: `True` when all functions have the required `operation` capability
+- **Failures**: Number of times the operation has failed and retried
+- **Pipeline**: Output from each function step for tracking progress
+- **`AppliedResourceRefs`**: References to all resources the Operation created or modified
+
+### Events
+
+Operations emit Kubernetes events for important activities:
+- Function run results and warnings
+- Resource apply failures
+- Operation lifecycle events (creation, completion, failure)
+
+### Troubleshooting operations
+
+**Select operation status:**
+
+```shell
+kubectl get operation my-operation -o wide
```

Comment:
should these be together in one prompt like they are? it looks a bit odd when rendered: https://deploy-preview-956--crossplane.netlify.app/master/operations/operation/#troubleshooting-operations